frontend/javascript/modules/newYear.js

```diff
@@ -0,0 +1,63 @@
+import FetchUtil from '../utils/fetchUtil';
+
+export default class NewYear {
+  constructor(link) {
+    this.link = link;
+    this.step = this.link.dataset.step;
+    this.uid = this.link.dataset.uid;
+
+    this.endpoints = {
+      housing: '/housing',
+      active: '/manage/active',
+      current: '/manage/current/'
+    };
+
+    this.render();
+  }
+  render() {
+    this.link.addEventListener('click', e => {
+      e.preventDefault();
+
+      if (this.step === "welcome") {
+        $('#new-welcome').fadeOut(function() {
+          $("#new-clear").fadeIn();
+        });
+      } else if (this.step === "clear") {
+        FetchUtil.fetchWithWarning(this.endpoints.active, {
+          method: 'DELETE',
+          warningText: "This will clear active members and room assignments!",
+          successText: "Data successfully cleared."}, () => {
+            fetch(this.endpoints.housing, {
+              method: 'DELETE'
+            })
+            .then($('#new-clear').fadeOut(function() {
+              $("#new-current").fadeIn();
+            })
+            );
+          });
+      } else if (this.uid) {
+        if ($('#rem-' + this.uid).is(":visible")) {
+          fetch(this.endpoints.current + this.uid, {
```

Comment:
Same as above. In addition, if you want the statements following to only be executed in the event of a successful request, they should be enclosed in a `.then()` on this promise.